lib/rubocop/cop/team.rb

```diff
@@ -174,11 +174,28 @@ def support_target_ruby_version?(cop)
       end
 
       def support_target_rails_version?(cop)
+        # In this case, the rails version was already checked by `#excluded_file?`
+        return true if rubocop_rails_uses_requires_gem?
+
         return true unless cop.class.respond_to?(:support_target_rails_version?)
 
         cop.class.support_target_rails_version?(cop.target_rails_version)
       end
 
+      # Returns true if the current version of rubocop-rails uses `requires_gem` to specify
+      # its gem requirements.
+      def rubocop_rails_uses_requires_gem?
+        @rubocop_rails_uses_requires_gem ||= begin
+          rubo_rails_version = RuboCop::Rails::Version::STRING if defined?(RuboCop::Rails)
+
+          if rubo_rails_version
+            Gem::Version.new('2.21.0') <= Gem::Version.new(rubo_rails_version)
```

Comment:
It looks like the versioning between `rubocop` and `rubocop-rails` gets a bit tricky here. 

In the future, `TargetRailsVersion` can just be a wrapper for `requires_gem`. When that happens, any cops that don't support the target's rails version would have already be filtered out by `#excluded_file?` called by `#roundup_relevant_cops` above. I'm assuming this will be done in version `2.21.0`.

For older versions of rubocop-rails, that won't be correct, and we'll still need to call their `support_target_rails_version?` method.